devtools/create_achievements/create_achievements.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/sh
+
+set -e
+
+trap "echo FAILURE: $0 failed to create achievements.dat" ERR
+
+usage() {
+    echo "Usage: $0 [OPTION]..."
+    echo "    -f, --force    Forced regeneration of all files"
+    echo "    -h, --help     Show usage text"
+    echo "    -v, --verbose  Verbose output"
+    exit 42
+}
+
+add_steam() {
+    if [[ "$FORCE" == "1" || ! -f "gen/steam-$1.ini" ]]; then
+        python steam_achievements.py ${VERBOSE:+"-v"} --steamid "$1"
```

Comment:
These are bash extensions, no? If so we should say /bin/bash - not /bin/sh (but I might be wrong here)

Reply:
@mgerhardy You are not wrong.
@lolbot-iichan : There is a tool to check for POSIX shell compliance called "checkbashisms": https://sourceforge.net/projects/checkbaskisms/

Reply:
Current output from my run is:
````
possible bashism in create_achievements.sh line 5 (trap with ERR|DEBUG|RETURN):
trap "echo FAILURE: $0 failed to create achievements.dat" ERR
possible bashism in create_achievements.sh line 16 (alternative test command ([[ foo ]] should be [ foo ])):
    if [[ "$FORCE" == "1" || ! -f "gen/steam-$1.ini" ]]; then
possible bashism in create_achievements.sh line 16 (should be 'b = a'):
    if [[ "$FORCE" == "1" || ! -f "gen/steam-$1.ini" ]]; then
possible bashism in create_achievements.sh line 23 (alternative test command ([[ foo ]] should be [ foo ])):
    if [[ "$FORCE" == "1" || ! -f "gen/galaxy-$2.ini" ]]; then
possible bashism in create_achievements.sh line 23 (should be 'b = a'):
    if [[ "$FORCE" == "1" || ! -f "gen/galaxy-$2.ini" ]]; then
possible bashism in create_achievements.sh line 42 (alternative test command ([[ foo ]] should be [ foo ])):
if [[ "$FORCE" == "1" ]]; then
possible bashism in create_achievements.sh line 42 (should be 'b = a'):
if [[ "$FORCE" == "1" ]]; then
possible bashism in create_achievements.sh line 133 (alternative test command ([[ foo ]] should be [ foo ])):
if [[ "$VERBOSE" == "-v" ]]; then
possible bashism in create_achievements.sh line 133 (should be 'b = a'):
if [[ "$VERBOSE" == "-v" ]]; then
````

Reply:
@mgerhardy Thanks for raising this. I always forget that sh != bash. Fixed at 2ee2335

@digitall Thanks for sharing an excellent tool. Used it to check commit before sending.

Reply:
No problem!